src/delagent/ui/admin-upload-delete.php

```diff
@@ -114,27 +115,79 @@ function Delete($uploadpk, $Depends = NULL)
   } // Delete()
 
   /**
+   * @param $uploadpks
+   * @brief starts deletion and handles error messages
+   * @return string
+   */
+  function initDeletion($uploadpks)
+  {
+    $V = "";
+    $errorMessages = [];
+    $deleteResponse = "";
+    for($i=0; $i < sizeof($uploadpks); $i++)
+    {
+      $deleteResponse = $this->TryToDelete($uploadpks[$i]);
```

Comment:
loosing the explicit cast to `int` makes this a security problem, especially with the unclean usage in `admin_upload_delete->Delete` (please also fix the `Delete` function).